bee-common/bee-common/src/event.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! A module that provides a generic, type-safe event bus for arbitrary event types.
+
+use dashmap::DashMap;
+use std::any::{Any, TypeId};
+
+type Listener<'a> = dyn Fn(&dyn Any) + Send + Sync + 'a;
+
+/// An event bus for arbitrary event types.
+pub struct Bus<'a, ID = TypeId> {
```

Comment:
Even if it's in the `event` module, I wonder if `Bus` is a sufficient name or not. When imported, I am not sure it's expressive enough. What do you think of `EventBus` ?

Reply:
I personally think it's expressive enough ("bus" is commonly used as some central communication system) but I can understand the opposing view. Is this something you'd like me to change?

Reply:
Not necessarily no, let's see what the others think

Reply:
I think I'ld either call it `EventBus` as well, or rename the module to `bus.rs`, and prevent to use the term `event` like here:
`pub fn dispatch<E: Any>(&self, event: E) {`.
Instead I'ld use `message` and the generic parameter `M`. We **use** this bus for an event system like we use a `Dashmap` here to store the listeners, but it's implementation is more generic, as Josh said, more like some sort of communication system, and the more general term for the thing that gets sent around in such a system is a `message`.

Reply:
I really don't have a strong opinion about this, but `EventBus` is more expressive and can be recognized better, so i am fine with "EventBus" if you guys want to use that.

> Instead I'ld use message and the generic parameter

i think the term `event` is generic enough to reflect this kind of communication systems (independent of low-level or high-level)
i would even prefer to avoid using the term message, since we have bee-message and  protocol-messages
and having bus-messages might even make more confusion

Reply:
long story short: there is no need to call it or the enclosing module `EventBus`/`event.rs` at all. We use it as such, but it could be used for something else as well.

Reply:
tbh, I am not in favor of using `message`. Look at this `Bus` implementation, where it's `val`. What this actually is here is some sort SPMC (broadcast) channel.
https://docs.rs/bus/2.2.3/src/bus/lib.rs.html#335

Reply:
So what is the conclusion of this discussion?